### PR TITLE
Goom Main.cpp Locking Fixes

### PR DESCRIPTION
**Pull request by glk1001**
Minor refactoring  and concurrency locking fixes.

### CODE REVIEW

File: src/AudioData.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  {
+    for (int i=0; i < AUDIO_SAMPLE_LEN; i++) 
+    {
+      audioData[0][i] = FloatToInt16(floatAudioData[i]);
+      audioData[1][i] = audioData[0][i];
+    }
+  } else 
+  {
+    int fpos = 0;
+    for (int i=0; i < AUDIO_SAMPLE_LEN; i++)
```
Comment (fritsch) [inline src/AudioData.h]: you could use i < AUDIO_SAMPLE_LEN / 2 and increase i by 2

Comment (glk1001) [reply src/AudioData.h]: Sorry, not sure what you mean. I was trying to keep this as simple as possible for clarity. Every second floatAudioData element goes to audioData[1][i]. I'm happy to change it if you show me the loop you mean.

Comment (fritsch) [reply src/AudioData.h]: Yes, sure:

```
for (int i = 0; i < AUDIO_SAMPLE_LEN / 2; i += 2)
{
  audiodata[0][i] = whatever[i];
  audiodate[1][i] = whatever[i+1];
}
```

Your code is fine - just decide what is easier.

Comment (fritsch) [reply src/AudioData.h]: You decide. If you say, your's is easier, please keep it and btw. thanks that you look into that stuff.

@AlwinEsch  please have a look at the locking.

File: src/Main.h [hunk 2]
```diff
@@ -65,15 +65,25 @@ class ATTRIBUTE_HIDDEN CVisualizationGoom
   void OnCompiledAndLinked() override;
   bool OnEnabled() override;
 
+protected:  
+  virtual void NoActiveBufferAvailable() {}
+  virtual void AudioDataQueueTooBig() {}
+  virtual void SkippedAudioData() {}
+  virtual void UpdateGoomBuffer(
```
Comment (AlwinEsch) [inline src/Main.h]: What was thought by add of this virtual functions?
Planned something about a child of this class?

File: src/AudioData.h (hunk 1 above)
Comment (AlwinEsch) [reply src/AudioData.h]: Also a good with yours example @fritsch

**Review by AlwinEsch — COMMENTED** [review]
The changes look good for me, I'll test it out on Sunday or Monday (I'm not home for the weekend).

Only thing that needs to be revised is the code style.

If the tests are OK, he could come in and Kodi Matrix Version too.

**Comment by AlwinEsch** [comment]
Any state about this? 

There another API related request now (https://github.com/xbmc/visualization.goom/pull/30) and want to bring that here in first.

**Comment by glk1001** [comment]
Hi Alwin,

Sorry for the late reply. I think for clarity I would like to keep the loop

    int fpos = 0;
    for (int i=0; i < AUDIO_SAMPLE_LEN; i++)
    {
      audioData[0][i] = FloatToInt16(floatAudioData[fpos]);
      fpos++;
      audioData[1][i] = FloatToInt16(floatAudioData[fpos]);
      fpos++;
    }

as is. It's tricky - floatAudioData has to loop through 2*AUDIO_SAMPLE_LEN
while audioData has to loop through AUDIO_SAMPLE_LEN.

Yes, the virtual methods

protected:
  virtual void NoActiveBufferAvailable() {}
  virtual void AudioDataQueueTooBig() {}
  virtual void SkippedAudioData() {}
  virtual void UpdateGoomBuffer(...

are for a child class I was using for testing. I can get rid of them if you
like.

What are the code style changes that need fixing. I can have them done by
tomorrow.

Thanks,
Greg

**Comment by AlwinEsch** [comment]
> What are the code style changes that need fixing. I can have them done by tomorrow.

You can make on extra last commit a cleanup about.
- Take e.g. this https://github.com/xbmc/audiodecoder.2sf/blob/Matrix/.clang-format and add to here
- Then run `clang-format -style=file -i src/*.[ch]*` 

That cleanup the whole code to match a style. 

I can also do this, but then can be come a lot of conflicts with your requests. Less work for you if in this request done.

Have added here https://github.com/xbmc/visualization.goom/pull/30/commits/f84924010c832bd81a1a5b6128eab0da07816643 a temporary commit how it looks then. Before was also not perfect, on yours was for me mainly about "line break" by "{".

A place there like `const GLfloat tempQuadData[]` to hold in previous style (like them after clang is ugly :smirk:) can then have a add of:
```cpp
// clang-format off
...
// clang-format on
```

**Comment by glk1001** [comment]
I added two extra commits - one removing the virtuals I use for testing and one with the clang format.
I like the idea of using clang-format (not sure about some of the rules but automatic format can't be argued with  :relaxed:).

**Comment by glk1001** [comment]
Sorry forgot to check compile. Maybe squash all this when you're happy with it.

**Comment by AlwinEsch** [comment]
Thanks for the changes! From commit history not the best :stuck_out_tongue_winking_eye:, but runtime test was OK.

Can not wait for cleanup, as the Kodi is already updated and takes then to long until new binaries are available.

**Comment by AlwinEsch** [comment]
Was a bit stupid, has not seen that this relate to Kodi Leia, has added your commits to https://github.com/xbmc/visualization.goom/pull/30 (bit cleaned to 2) and then also on Matrix.

The runtime tests with them was good.

**Pull request merged**